services/workflows-service/src/alert/alert.service.ts

```diff
@@ -140,8 +152,56 @@ export class AlertService {
     }
   }
 
-  // Specific alert check logic based on the definition
-  private async checkAlert(alertDefinition: AlertDefinition) {
+  async checkOngoingMonitoringAlert(businessReport: BusinessReport, businessCompanyName: string) {
+    const alertDefinitions = await this.alertDefinitionRepository.findMany(
```

Comment:
you can use this function just add additional projectId optional filter to `getAlertDefinitions `
```
    const alertDefinitions = await this.getAlertDefinitions({
      type: MonitoringType.transaction_monitoring,
    });
```

Reply:
I understand, but it's being handled by the repository, using scoping.

why do you need to do it twice?

Reply:
ok so change the previous method to use the repository as well